calico_kubernetes/calico_kubernetes.py

```diff
@@ -192,29 +229,63 @@ def status(self, namespace, pod_name, docker_id):
 
-        if self._datastore_client.profile_exists(self.profile_name):
-            logger.error("Profile with name %s already exists, exiting.",
-                         self.profile_name)
-            sys.exit(1)
+        Otherwise, the pod gets assigned to the default profile.
+        """
+        if self.default_policy != POLICY_NONE:
+            # Determine the name for this profile.
+            profile_name = "%s_%s_%s" % (self.namespace, 
+                                         self.pod_name, 
+                                         str(self.docker_id)[:12])
+
+            # Create a new profile for this pod.
+            logger.info("Creating profile '%s'", profile_name)
+
+            #  Retrieve pod labels, etc.
+            pod = self._get_pod_config()
+    
+            if self._datastore_client.profile_exists(profile_name):
+                # In profile-per-pod, we don't ever expect duplicate profiles.
+                logger.error("Profile '%s' already exists.", profile_name)
+                sys.exit(1)
+            else:
+                # The profile doesn't exist - generate the rule set for this 
+                # profile, and create it.
+                rules = self._generate_rules(pod, profile_name)
+                self._datastore_client.create_profile(profile_name, rules)
+    
+            # Add tags to the profile based on labels.
+            self._apply_tags(pod, profile_name)
+    
+            # Set the profile for the workload.
+            logger.info("Setting profile '%s' on endpoint %s",
+                        profile_name, endpoint.endpoint_id)
+            self._datastore_client.set_profiles_on_endpoint(
+                [profile_name], endpoint_id=endpoint.endpoint_id
+            )
+            logger.debug('Finished configuring profile.')
         else:
-            rules = self._generate_rules(pod)
-            self._datastore_client.create_profile(self.profile_name, rules)
-
-        # Add tags to the profile.
-        self._apply_tags(pod)
-
-        # Set the profile for the workload.
-        logger.info('Setting profile %s on endpoint %s',
-                    self.profile_name, endpoint.endpoint_id)
-        self._datastore_client.set_profiles_on_endpoint(
-            [self.profile_name], endpoint_id=endpoint.endpoint_id
-        )
-        logger.debug('Finished configuring profile.')
+            # Policy is disabled - add this pod to the default profile.
+            if not self._datastore_client.profile_exists(DEFAULT_PROFILE_NAME):
+                # If the default profile doesn't exist, create it.
+                logger.info("Creating profile '%s'", DEFAULT_PROFILE_NAME)
+                allow = Rule(action="allow")
+                rules = Rules(id=DEFAULT_PROFILE_NAME, 
+                              inbound_rules=[allow], 
+                              outbound_rules=[allow])
+                self._datastore_client.create_profile(DEFAULT_PROFILE_NAME, 
```

Comment:
Paranoid case: we check for the profile, it doesn't exist, so we try to create it. In the meantime, another process creates it. Will this call fail and take the whole function call down? Could we be more defensive, and re-check for existence if we fail here? If we fail and then the profile is there, we can continue.